File: src/main/java/com/history/tree/controllers/TestController.java
package com.history.tree.controllers;

import com.history.tree.model.Tree;
import com.history.tree.services.TreeTestDataGenerator;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

@RestController
@RequiredArgsConstructor
@RequestMapping("/debug")
public class TestController {

    private final TreeTestDataGenerator treeTestDataGenerator;

    @PostMapping("create_test_tree")
    public Mono<Tree> createTestTree() {
        return treeTestDataGenerator.generate();
    }
}

File: src/main/java/com/history/tree/controllers/TestController.kt
package com.history.tree.controllers

import com.history.tree.model.Tree
import com.history.tree.model.User
import com.history.tree.services.TreeTestDataGenerator
import kotlinx.coroutines.flow.Flow
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/debug")
class TestController(val treeTestDataGenerator: TreeTestDataGenerator) {

    @PostMapping("create_test_tree")
    suspend fun createTestTree(): Tree {
        return treeTestDataGenerator.generate()
    }

    @PostMapping("create_test_users")
    suspend fun createTestUsers(): Flow<User> {
        return treeTestDataGenerator.createTestsUsers()
    }
}